features/support/check/checkHasFocus.js

```diff
@@ -0,0 +1,19 @@
+const assert = require('assert').strict;
+
+/**
+ * Checks if a given element is an active element in the DOM.  Fails if the element does not exist.
+ * @param {String} selector CSS selector of the element to check for child elements.
+ * @param {String} not The string "not" to negate the check (the element should not have child elements).
+ */
+module.exports = async function(selector, not) {
+  const childFocus = await this.page.$(selector);
```

Comment:
The check you've added looks good, but you could shorten up the logic by using a similar approach to [checkContainsText.js](https://github.com/patheard/cucumber-puppeteer/blob/master/features/support/check/checkContainsText.js).  

```suggestion
/* istanbul ignore next */  // Required otherwise code coverage evaluation fails within $eval calls
const hasFocus= await this.page.$eval(selector, el => el === document.activeElement);
const shouldBeFocus = not ? false : true;

assert.strictEqual(hasFocus, shouldBeFocus, `Expected "${selector}" to ${shouldBeFocus ? 'have focus' : 'have no focus'}`);

```

The reason this works is because `page.$eval` throws an error if the selector doesn't exist, so you could remove your `if/else` and you'd just need to update your unit tests since it would be a different error coming back.